### PR TITLE
Ianhelle/2020 07 03 splunk connector

### PR DESCRIPTION
**Pull request by ianhelle**
Splunk connector updates

### CODE REVIEW

File: msticpy/data/query_source.py [hunk 1]
```diff
@@ -82,9 +104,11 @@ def __init__(self, name: str, source: dict, defaults: dict, metadata: dict):
         self.params = ChainMap(
             _value_or_default(self._source, "parameters", {}),
             _value_or_default(self.defaults, "parameters", {}),
+            # self._source.get("parameters", {}),
+            # self.defaults.get("parameters", {}),
```
Comment (petebryan) [inline msticpy/data/query_source.py:108]: Were these meant to be left in?

File: msticpy/data/drivers/splunk_driver.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+            Query results in a dataframe.
+            or query response if an error.
+
+        """
+        del query_source
+        if not self._connected:
+            raise MsticpyNotConnectedError(
+                "Please run the connect() method before running this method.",
+                title="not connected to Splunk.",
+                help_uri="TBD",
```
[the remark this answers is not in the document]
Comment (ianhelle) [reply msticpy/data/drivers/splunk_driver.py:188]: Yeah - we need some documents to point to.

File: msticpy/data/drivers/splunk_driver.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        self, query: str, query_source: QuerySource = None, **kwargs
+    ) -> Union[pd.DataFrame, Any]:
+        """
+        Execute splunk query and retrieve results via OneShot search mode.
+
+        Parameters
+        ----------
+        query : str
+            Splunk query to execute via OneShot search mode
+        query_source : QuerySource
```
[the remark this answers is not in the document]
Comment (ianhelle) [reply msticpy/data/drivers/splunk_driver.py:167]: If I remove it I get a warning but I should change the doc to say "not used" or something.

File: msticpy/data/query_source.py (hunk 1 above)
Comment (ianhelle) [reply msticpy/data/query_source.py:108]: No - good spot.

File: msticpy/data/drivers/splunk_driver.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+
+        """
+        del query_source
+        if not self._connected:
+            raise MsticpyNotConnectedError(
+                "Please run the connect() method before running this method.",
+                title="not connected to Splunk.",
+                help_uri="TBD",
+            )
+        # If query is prefixed with "search", add that.
```
[the remark this answers is not in the document]
Comment (ianhelle) [reply msticpy/data/drivers/splunk_driver.py]: Deleted this

File: msticpy/data/drivers/splunk_driver.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ianhelle) [reply msticpy/data/drivers/splunk_driver.py]: Updated with generic URL to Dataproviders doc

**Pull request merged**